src/components/molecules/ChatsList/ChatsList.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useContext, useMemo, useState } from "react";
+
+// Components
+import {
+  ChatContext,
+  PrivateChatMessage,
+} from "components/context/ChatContext";
+import UserProfilePicture from "components/molecules/UserProfilePicture";
+import ChatBox from "components/molecules/Chatbox";
+
+// Hooks
+import { useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+
+// Utils | Settings | Constants
+import { DEFAULT_PARTY_NAME, VENUE_CHAT_AGE_DAYS } from "settings";
+import { getDaysAgoInSeconds } from "utils/time";
+import { WithId } from "utils/id";
+
+// Typings
+import { User } from "types/User";
+
+// Styles
+import "./ChatsList.scss";
+import { setPrivateChatMessageIsRead } from "components/organisms/PrivateChatModal/helpers";
+import { isEmpty } from "lodash";
+
+interface LastMessageByUser {
+  [userId: string]: PrivateChatMessage;
+}
+
+const ChatsList: React.FunctionComponent = () => {
+  const { user } = useUser();
+  const { privateChats, chatUsers } = useSelector((state) => ({
+    privateChats: state.firestore.ordered.privatechats,
+    chatUsers: state.firestore.data.chatUsers,
+  }));
+
+  const [selectedUser, setSelectedUser] = useState<WithId<User>>();
+
+  const discussionPartnerWithLastMessageExchanged =
+    privateChats &&
+    privateChats.reduce<LastMessageByUser>((agg, item) => {
+      let lastMessageTimeStamp;
+      let discussionPartner;
+      if (item.from === user?.uid) {
+        discussionPartner = item.to;
+      } else {
+        discussionPartner = item.from;
+      }
+      try {
+        lastMessageTimeStamp = agg[discussionPartner].ts_utc;
+        if (lastMessageTimeStamp < item.ts_utc) {
+          agg[discussionPartner] = item;
+        }
+      } catch {
+        agg[discussionPartner] = item;
+      }
+      return agg;
+    }, {});
```

Comment:
I would generally extract `reducer`, `filter`, `sort`, etc functions that don't need to 'close around' component state into static `const`s outside of the component, eg. `const fooReducer = (agg, item) = > /* ... */`

This is a 'free' way to get similar effects to memoisation for it (eg. `useMemo`, `useCallback`)

Though in this case you probably also want to memo `discussionPartnerWithLastMessageExchanged` to avoid having to re-filter it on every render.

Something like:

```suggestion
  const discussionPartnerWithLastMessageExchanged = useMemo(() => {
    if (!privateChats) return ({})
    return privateChats.reduce<LastMessageByUser>(lastMessageByUserReducer, {})
  }, [privateChats])
```

Reply:
fixed in 77fc5d6d1829a42da2152d00871cde0b6f47269d